### ISSUE TITLE
[Task]: Handle Empty regex in Ballerina

### ISSUE THREAD

**Issue by SasinduDilshara** [ballerina-platform/ballerina-lang#39796]
### Description

- [] #39795
- [] #39794

### Related area

-> Compilation

**Comment by SasinduDilshara** [comment]
So the discussed plan is to handle empty regex in following way.

Scenario 1

```
public function main() {
    string:RegExp b = re ``;
}
```

This will gives a compile time error. Ballerina will not allow to define empty regex in compile time.

Scenario 2:

```
public function main() {
    string a = "";
    string:RegExp b = re `abc${""}`;
    string:RegExp c = re `abc${a}`;
}
```

These codes are handle in runtime. The regex pattern will replace with non capturing group in runtime.
So the runtime convert `abc${""}` to `?:` and execute the code.

**Comment by sanjana** [comment]
The first task is addressed in #39848.

**Comment by sanjana** [comment]
Fixed in #39848
